modules/trainer/GenericTrainer.py

```diff
@@ -121,6 +122,13 @@ def start(self):
             else:
                 print("No backup found, continuing without backup...")
 
+        if self.config.secrets.huggingface_token != "":
+            self.callbacks.on_update_status("logging into Hugging Face")
+            huggingface_hub.login(
```

Comment:
not 100% if here is the best point to log in, but I think so.
in `HFModelLoaderMixin` there is no train config available there